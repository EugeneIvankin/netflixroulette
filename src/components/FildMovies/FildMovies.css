.main{
    display: flex;
    flex-flow: row wrap;
    justify-content: center;
}

nav{
    display: flex;
    align-items: center;
    justify-content: flex-end;
    height: 5vh;
    background-color: #f5f5f5;
}

.movies-found-text, .sort-by-text, #sort-by-date-text, #sort-by-rating-text {
    font-family: Arial;
    cursor: default;
    margin: 0;
    color: #414141;
}

.movies-found-text {
    margin-left: 7vw;
    margin-right: auto;
}

.sort-by-text, #sort-by-date-text, #sort-by-rating-text {
    margin-right: 7vw;
}

#sort-by-date-text:hover, #sort-by-rating-text:hover{
    cursor: pointer;
    color: #f55263;
}