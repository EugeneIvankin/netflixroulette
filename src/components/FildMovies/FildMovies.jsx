import React, { Component } from 'react';
import PropTypes from 'prop-types';
import Movie from '../Movie/Movie';
import NotFoundMovies from '../MovieNotFounded/NotFound';
import style from './FildMovies.css';


export default class MoviesSearch extends Component {
  renderList() {
    const { foundedMovies, foundedMovie } = this.props;
    if (foundedMovies && foundedMovies.length !== 0) {
      return foundedMovies
        .filter(item => !foundedMovie || foundedMovie.movieId !== item.movieId)
        .map(item => <Movie item={item} key={item.movieId} />);
    }
    return <NotFoundMovies />;
  }

  render() {
    return (
      <main>
        <div className={style.fildMovies}>
          {this.renderList()}
        </div>
      </main>
    );
  }
}

MoviesSearch.propTypes = {
  foundedMovies: PropTypes.arrayOf(PropTypes.object).isRequired,
  foundedMovie: PropTypes.shape({
    movieId: PropTypes.number,
  }),
};

MoviesSearch.defaultProps = {
  foundedMovie: {
    movieId: '',
  },
};
