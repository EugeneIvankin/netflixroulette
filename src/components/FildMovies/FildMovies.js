import React, { Component } from 'react'
import './FildMovies.css'
import PropTypes from 'prop-types'
import Movie from '../Movie/Movie'
import NotFoundMovies from '../MovieNotFounded/NotFound'


export default class MoviesSearch extends Component {

    constructor(props){
        super(props);
        this.state = {value:''};
       /* this.getMovie = this.getMovie.bind(this);*/
    }

   /* changeColorByClick(sort){
        switch (sort) {
            case 'date':
                document.getElementById('sort-by-date-text').style.color = '#f55263';
                document.getElementById('sort-by-rating-text').style.color = '#414141';
                break;
            case 'rating':
                document.getElementById('sort-by-date-text').style.color = '#414141';
                document.getElementById('sort-by-rating-text').style.color = '#f55263';
                break;
            default:
                document.getElementById('sort-by-date-text').style.color = '#414141';
                document.getElementById('sort-by-rating-text').style.color = '#414141';
        }
    }*/

    getMovie = (movieId) =>{
        this.props.history.push(`/film/${movieId}`);
        window.scrollTo(0,0);
    }

    renderList(){
        const {foundedMovies, foundedMovie} = this.props;
        if (foundedMovies.length !==0){
            return foundedMovies
                .filter(item=>!foundedMovie||foundedMovie.movieId!==item.movieId)
                .map((item) =>
                   <Movie item={item} getMovie={this.getMovie} key={item.movieId}/>
                );
        }
        return <NotFoundMovies/>
    }


    // вынести значения сортировки в стейт

    sortByDate = () => {
        const {sortMovies} = this.props;
        sortMovies('date')
    }

    sortByRating = () =>{
        const {sortMovies} = this.props;
        sortMovies('rating')
    }

    //

    render() {
        const {foundedMovies} = this.props;

        return <main>
            <nav>
                <p className="movies-found-text"> {foundedMovies.length} movies found </p>
                <p className="sort-by-text"> Sort by </p>
                <p id="sort-by-date-text" onClick={this.sortByDate}> realease date</p>
                <p id="sort-by-rating-text" onClick={this.sortByRating}> rating </p>
            </nav>
            <div className="main">
                {this.renderList()}
            </div>
        </main>
    }
}

// имена
// вынести скрол в роут
// стили
// проверить в конструкторе стейт
// убрать лямда
// вынести лист


MoviesSearch.propTypes = {
  foudedMovies: PropTypes.array, foundedMovie: PropTypes.object, sortMovies: PropTypes.func
};


