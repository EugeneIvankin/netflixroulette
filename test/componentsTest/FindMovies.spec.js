import React from 'react';
import { shallow } from 'enzyme';
import FildMovies from '../../src/components/FildMovies/FildMovies'

describe('<FildMovies/>', () => {

    const foundedMovies = [
        {movieId: '1', movieName: 'Taxi', moviePoster: 'url', movieYear: '2017', movieLevel: '5.5'}
    ];
    const foundedMovie = {
        movieId: '1',
        movieName: 'Taxi',
        moviePoster: 'url',
        movieYear: '2017',
        movieLevel: '5.5'
    };

    it ('should render FindMovies with foundedMovies', () => {
        const renderedComponent = shallow(
          <FildMovies foundedMovies={foundedMovies}
                      foundedMovie={foundedMovie}
                      sortMovies={jest.fn()}/>
        );
        expect(renderedComponent).toMatchSnapshot();
    });

    it ('should render FindMovies without foundedMovies', () => {
        const renderedComponent = shallow(
            <FildMovies foundedMovies={[]}
                        foundedMovie={foundedMovie}
                        sortMovies={jest.fn()}/>
        );
        expect(renderedComponent).toMatchSnapshot();
    });

    it ('should call getMovie', () => {
        const history = {
            push: jest.fn()
        };

        const renderedComponent = shallow(
            <FildMovies history={history}
                        foundedMovies={foundedMovies}
                        foundedMovie={foundedMovie}/>
        );

        renderedComponent.instance().getMovie(13);
        expect(history.push).toBeCalledWith('/film/13');
    })

});