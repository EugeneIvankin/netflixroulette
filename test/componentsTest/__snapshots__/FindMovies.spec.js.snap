// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<FildMovies/> should render FindMovies with foundedMovies 1`] = `
<main>
  <div
    className="fild-movies"
  />
</main>
`;

exports[`<FildMovies/> should render FindMovies without foundedMovies 1`] = `
<main>
  <div
    className="fild-movies"
  >
    <NotFoundPage />
  </div>
</main>
`;
